walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/serializer/PresentationExchangeResponseBuilder.kt

```diff
@@ -38,78 +38,70 @@ internal class PresentationExchangeResponseBuilder(
                 ttlInSeconds,
                 audience,
                 nonce
             )
         }
     }
-    suspend fun buildIdToken(definitionId: String,
-                             clientId: String,
-                             requestNonce: String,
-                             ttlInSeconds: Int = 3600): String {
-        val exception = libraryConfiguration.identifierManager.getMasterIdentifier().toNative().map { identifier ->
-            val (issuedTime, expiryTime) = createIssuedAndExpiryTime(ttlInSeconds)
-            val vpTokens = this.vpTokens.mapIndexed {
-                index, vpToken ->
-                vpToken.getPresentationSubmissionMap(index)
-            }.flatten()
 
-            val submission = VpTokenInResponse(
-                PresentationSubmission(
-                    id = UUID.randomUUID().toString(),
-                    definitionId = definitionId,
-                    presentationSubmissionDescriptors = vpTokens
-                )
-            )
-
-            val vpClaims = PresentationResponseClaims(listOf(submission))
+    suspend fun buildIdToken(
+        definitionId: String,
+        clientId: String,
+        requestNonce: String,
+        ttlInSeconds: Int = 3600
+    ): String {
+        val identifier = libraryConfiguration.identifierFactory.getIdentifier()
+        val (issuedTime, expiryTime) = createIssuedAndExpiryTime(ttlInSeconds)
+        val vpTokens = this.vpTokens.mapIndexed { index, vpToken ->
+            vpToken.getPresentationSubmissionMap(index)
+        }.flatten()
 
-            val oidcResponseClaims = vpClaims.apply {
-                subject = identifier.id
-                audience = clientId
-                nonce = requestNonce
-                responseCreationTime = issuedTime
-                responseExpirationTime = expiryTime
-            }
+        val submission = VpTokenInResponse(
+            PresentationSubmission(
+                id = UUID.randomUUID().toString(),
+                definitionId = definitionId,
+                presentationSubmissionDescriptors = vpTokens
+            )
+        )
 
-            val token = libraryConfiguration.serializer.encodeToString(PresentationResponseClaims.serializer(), oidcResponseClaims)
-            return libraryConfiguration.tokenSigner.signWithIdentifier(token, identifier)
-        }.exceptionOrNull()
+        val vpClaims = PresentationResponseClaims(listOf(submission))
 
-        if (exception != null) {
-            throw exception
+        val oidcResponseClaims = vpClaims.apply {
+            subject = identifier.id
+            audience = clientId
+            nonce = requestNonce
+            responseCreationTime = issuedTime
+            responseExpirationTime = expiryTime
         }
-        // unreachable
-        return ""
+
+        val token = libraryConfiguration.serializer.encodeToString(PresentationResponseClaims.serializer(), oidcResponseClaims)
+        return identifier.sign(token)
```

Comment:
I don't think this returns the expected value. Comparing with iOS, I think we may be returning different results for the HolderIdentifier's `sign` function. It seems we're returning a complete JWS token here, but iOS returns just the signature.

Reply:
Changed the implementation to return only the signature.